packages/wallet-sdk/src/sign/scw/utils/createSmartAccount.ts

```diff
@@ -0,0 +1,830 @@
+  if (owner.type === 'webAuthn') {
+    const { signature, webauthn } = await owner.sign({
+      hash,
+    });
+    return toWebAuthnSignature({ signature, webauthn });
+  }
+
+  if (owner.sign) return owner.sign({ hash });
+
+  throw new BaseError('`owner` does not support raw sign.');
+}
+
+/** @internal */
+export function toReplaySafeHash({
+  address,
+  chainId,
+  hash,
+}: {
+  address: Address;
+  chainId: number;
+  hash: Hash;
+}) {
+  return hashTypedData({
+    domain: {
+      chainId,
+      name: 'Coinbase Smart Wallet',
+      verifyingContract: address,
+      version: '1',
+    },
+    types: {
+      CoinbaseSmartWalletMessage: [
+        {
+          name: 'hash',
+          type: 'bytes32',
+        },
+      ],
+    },
+    primaryType: 'CoinbaseSmartWalletMessage',
+    message: {
+      hash,
+    },
+  });
+}
+
+/** @internal */
+export function toWebAuthnSignature({
+  webauthn,
+  signature,
+}: {
+  webauthn: WebAuthnP256.SignMetadata;
+  signature: Hex;
+}) {
+  const { r, s } = Signature.fromHex(signature);
+  return encodeAbiParameters(
+    [
+      {
+        components: [
+          {
+            name: 'authenticatorData',
+            type: 'bytes',
+          },
+          { name: 'clientDataJSON', type: 'bytes' },
+          { name: 'challengeIndex', type: 'uint256' },
+          { name: 'typeIndex', type: 'uint256' },
+          {
+            name: 'r',
+            type: 'uint256',
+          },
+          {
+            name: 's',
+            type: 'uint256',
+          },
+        ],
+        type: 'tuple',
+      },
+    ],
+    [
+      {
+        authenticatorData: webauthn.authenticatorData,
+        clientDataJSON: stringToHex(webauthn.clientDataJSON),
+        challengeIndex: BigInt(webauthn.challengeIndex),
+        typeIndex: BigInt(webauthn.typeIndex),
+        r,
+        s,
+      },
+    ]
+  );
+}
+
+/** @internal */
+export function wrapSignature(parameters: { accountIndex?: number | undefined; signature: Hex }) {
+  const { accountIndex = 0 } = parameters;
+  const signatureData = (() => {
+    if (size(parameters.signature) !== 65) return parameters.signature;
+    const signature = parseSignature(parameters.signature);
+    return encodePacked(
+      ['bytes32', 'bytes32', 'uint8'],
+      [signature.r, signature.s, signature.yParity === 0 ? 27 : 28]
+    );
+  })();
+  return encodeAbiParameters(
+    [
+      {
+        components: [
+          {
+            name: 'accountIndex',
+            type: 'uint8',
+          },
+          {
+            name: 'signatureData',
+            type: 'bytes',
+          },
+        ],
+        type: 'tuple',
+      },
+    ],
+    [
+      {
+        accountIndex,
+        signatureData,
+      },
+    ]
+  );
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////////
+// Constants
+/////////////////////////////////////////////////////////////////////////////////////////////
+
+const abi = [
```

Comment:
do we already have this ABI in the constants file?

Reply:
Yes. Thanks for flagging